server/utils/getSearchOptions.ts

```diff
@@ -0,0 +1,14 @@
+import type { Request } from 'express'
+
+export const getSearchOptions = <T>(request: Request, keys: Array<keyof T>): T => {
+  const searchOptions = {} as T
```

Comment:
If we're casting on line 4 then I think we're best off having `keys` as an `Array<string>` which would mean we don't have to cast on line 7 and 9